boards: infineon: add trade marks to board names

yurii2001lozyn opened the pull request:
Update board full names to include PSOC™ trade mark symbols for consistent branding across Infineon board definitions.

lozyn888 committed:
boards: infineon: add trade marks to board names

Update board full names to include PSOC™ trade mark symbols
for consistent branding across Infineon board definitions.

Signed-off-by: Yurii Lozynskyi <yurii.lozynskyi@infineon.com>

yurii2001lozyn commented on boards/infineon/cy8ckit_062s2_ai/board.yml:
```diff
@@ -3,7 +3,7 @@
 board:
   name: cy8ckit_062s2_ai
-  full_name: PSOC 6 AI Evaluation Kit
+  full_name: PSOC™ 6 AI Evaluation Kit
```
Okay I changed ™ to  |trade| but It doesn't render correctly 

<img width="442" height="382" alt="image" src="https://github.com/user-attachments/assets/f43affe2-ddca-409e-b3f0-9d938dfa7906" />

JarmouniA commented on samples/drivers/uart/async_api/boards/cy8ckit_062s2_ai.overlay:
unrelated change

JarmouniA commented on tests/drivers/uart/uart_async_api/testcase.yaml:
```diff
@@ -3,6 +3,7 @@ common:
     - stamp_c3
     - wio_terminal
     - xiao_esp32c3
+    - cy8ckit_062s4
```
ditto

JarmouniA reviewed (dismissed):
If the sample changes are needed for CI, put them in a separate commit.

lozyn888 committed:
uart: async_api: add cy8ckit_062s2_ai board support

Add overlay files and sample configuration for cy8ckit_062s2_ai
to enable UART async API functionality.

Signed-off-by: Yurii Lozynskyi <yurii.lozynskyi@infineon.com>

kartben commented on drivers/serial/Kconfig.ifx_cat1:
```diff
@@ -10,7 +10,8 @@ config UART_INFINEON_CAT1
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
-	select SERIAL_SUPPORT_ASYNC if !SOC_FAMILY_INFINEON_CAT1C
+	select SERIAL_SUPPORT_ASYNC if !(SOC_FAMILY_INFINEON_CAT1C || \
```
I really doubt you want to filter based on a _board_?

yurii2001lozyn commented on samples/drivers/uart/async_api/sample.yaml:
```diff
@@ -4,6 +4,9 @@ tests:
       - nrf52840dk/nrf52840
+    platform_exclude:
+      - cy8ckit_062s4
+      - cy8ckit_062s2_ai
```
Added filter for sample and updated ifx Kconfig

JarmouniA reviewed (commented):
2nd commit message is outdated.

JarmouniA reviewed (commented):
Also, you should put the driver's Kconfig change in a separate commit.

lozyn888 committed:
drivers: uart: ifx_cat1: disable async API for SOC_CY8C6244LQI_S4D92

Exclude SOC_CY8C6244LQI_S4D92 from UART async API support due to
hardware limitations, similar to SOC_FAMILY_INFINEON_CAT1C.

Signed-off-by: Yurii Lozynskyi <yurii.lozynskyi@infineon.com>

kartben commented on samples/drivers/uart/async_api/sample.yaml:
```diff
@@ -9,6 +9,7 @@ tests:
       - uart
     filter: CONFIG_SERIAL and
             CONFIG_UART_ASYNC_API and
+            CONFIG_SERIAL_SUPPORT_ASYNC and
```
CONFIG_UART_ASYNC_API already depends on SERIAL_SUPPORT_ASYNC

yurii2001lozyn replied:
Done
